File: video/getting-started/use-participant-media-ios/use-participant-media-ios.m
// Set the Participant Delegate to receive participant's events
participant.delegate = self;

// TVIVideoView where you want to render remote participant's video
@property (nonatomic, weak) IBOutlet TVIVideoView *remoteView;

#pragma mark - TVIParticipantDelegate

- (void)participant:(nonnull TVIParticipant *)participant addedVideoTrack:(nonnull TVIVideoTrack *)videoTrack {
    // Add remoteView on video track to render video
    [videoTrack addRenderer:self.remoteView];
}

- (void)participant:(nonnull TVIParticipant *)participant removedVideoTrack:(nonnull TVIVideoTrack *)videoTrack {
    // To stop rendering simply call removeRenderer
    [videoTrack removeRenderer:self.remoteView];
}
- (void)participant:(nonnull TVIParticipant *)participant addedAudioTrack:(nonnull TVIAudioTrack *)audioTrack {
    // A Participant added an Audio Track
}

- (void)participant:(nonnull TVIParticipant *)participant removedAudioTrack:(nonnull TVIAudioTrack *)audioTrack {
    // A Participant removed an Audio Track
}

- (void)participant:(nonnull TVIParticipant *)participant enabledTrack:(nonnull TVITrack *)track {
    // A Participant enabled a Track
}

- (void)participant:(nonnull TVIParticipant *)participant disabledTrack:(nonnull TVITrack *)track {
    // A Participant disabled a Track
}

File: video/getting-started/use-participant-media-ios/use-participant-media-ios.swift
// Set the Participant Delegate to receive participant's events
participant.delegate = self;

// TVIVideoView where you want to render remote participant's video
@IBOutlet weak var remoteView: TVIVideoView!

// MARK: TVIParticipantDelegate

func participant(_ participant: TVIParticipant, addedVideoTrack videoTrack: TVIVideoTrack) {
    // Add remoteView on video track to render video
    videoTrack.addRenderer(remoteView);
}

func participant(_ participant: TVIParticipant, removedVideoTrack videoTrack: TVIVideoTrack) {
    // To stop rendering simply call removeRenderer
    videoTrack.removeRenderer(remoteView);
}

func participant(_ participant: TVIParticipant, addedAudioTrack audioTrack: TVIAudioTrack) {
    // A participant added an Audio Track
}

func participant(_ participant: TVIParticipant, removedAudioTrack audioTrack: TVIAudioTrack) {
    // A participant removed an Audio Track
}

func participant(_ participant: TVIParticipant, enabledTrack track: TVITrack) {
    // A participant enabled a Track
}

func participant(_ participant: TVIParticipant, disabledTrack track: TVITrack) {
    // A participant disabled a Track
}
